### PR TITLE
Executor ITF: query permissions common test

### PR DESCRIPTION
**Pull request by MBoldyrev**
Signed-off-by: Mikhail Boldyrev <miboldyrev@gmail.com>

### Description of the Change

This adds an implementation of common base fixture - a parametric test class for query permission testing on the new shiny Executor ITF.

### Usage Examples or Tests *[optional]*

All you need to create a permission test using this class is:
https://github.com/hyperledger/iroha/blob/b01dc38a6a58077a9cd7ee7343a8814da660bb65/test/integration/executor/get_signatories_test.cpp#L94-L112

### CODE REVIEW

File: test/integration/executor/query_permission_test.hpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        // create spectators
+        assertResultValue(getItf().createUserWithPerms(
+            kAnotherUser,
+            kDomain,
+            kSameDomainUserKeypair.publicKey(),
+            permissions_param_.spectator_permissions));
+        assertResultValue(getItf().createUserWithPerms(
+            kAnotherUser,
+            kSecondDomain,
+            kAnotherDomainUserKeypair.publicKey(),
```
Comment (lebdron) [inline test/integration/executor/query_permission_test.hpp]: I am not sure if `second` or `another` is better, but the suggestion is to make them consistent - either use `second`, or `another`.

File: test/integration/executor/query_permission_test.hpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        shared_model::interface::RolePermissionSet
+            permission_to_query_my_domain,
+        shared_model::interface::RolePermissionSet
+            permission_to_query_everyone);
+
+    template <typename SpecificQueryFixture>
+    struct QueryPermissionTest
+        : public SpecificQueryFixture,
+          public ::testing::WithParamInterface<
+              std::tuple<std::shared_ptr<ExecutorTestParam>,
```
[the remark this answers is not in the document]
Comment (MBoldyrev) [reply test/integration/executor/query_permission_test.hpp]: google test manages parameter combinations via structs, so I'd conform to it

File: test/integration/executor/query_permission_test.hpp [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      }
+
+      /**
+       * Prepare state of ledger:
+       * - create accounts of target user, close and remote spectators. Close
+       *   spectator is another user from the same domain as the domain of
+       * target user account, remote - a user from domain different to domain
+       * of target user account.
+       *
+       * @param target_permissions - set of permisisons for target user
```
Comment (luckychess) [inline test/integration/executor/query_permission_test.hpp]: typo `permissions`

File: test/integration/executor/query_permission_test.cpp [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+#include "integration/executor/query_permission_test.hpp"
+
+#include "framework/common_constants.hpp"
+
+using namespace common_constants;
+using namespace executor_testing;
+
+using shared_model::interface::RolePermissionSet;
+
+enum class SpectatorPermissions : int {
```
Comment (luckychess) [inline test/integration/executor/query_permission_test.cpp]: Why we ask for `int` explicitly here?

**Pull request merged**